metadata-db-app/test/cmr/metadata_db/test/services/subscriptions_test.clj

```diff
@@ -26,102 +34,177 @@
+                                                                                                                  :endpoint "some-endpoint"
+                                                                                                                  :method "ingest"
+                                                                                                                  :mode ["New"]
+                                                                                                                  :collection-concept-id "C12345-PROV1"}
+                                                                                                   :concept-type :subscription}))}
+        (is (= 1 (subscriptions/change-subscription test-context :subscription {:extra-fields {:subscription-type "granule"
+                                                                                               :endpoint "some-endpoint"
+                                                                                               :method "ingest"
+                                                                                               :mode ["New"]
+                                                                                               :collection-concept-id "C12345-PROV1"}})))))
     (testing "Delete a subscription"
-      (is (= 1 (subscriptions/delete-subscription test-context :subscription {:extra-fields {:collection-concept-id "C12345-PROV1"
-                                                                                             :mode ["New"]
-                                                                                             :endpoint "some-endpoint"
-                                                                                             :method "ingest"}}))))
+      (with-bindings {#'subscriptions/get-subscriptions-from-db (fn [_context _coll-concept-id] '())}
+        (is (= 1 (subscriptions/change-subscription test-context :subscription {:extra-fields {:collection-concept-id "C12345-PROV1"
+                                                                                               :mode ["New"]
+                                                                                               :endpoint "some-endpoint"
+                                                                                               :method "ingest"}})))))
+    (testing "Add-to-existing-mode"
+      (are3
+       [expected existing-modes new-modes]
+       (is (= expected (subscriptions/add-to-existing-mode existing-modes new-modes)))
+
+       "new subscription with no existing modes in cache."
+       ["New" "Update" "Delete"]
+       nil
+       ["New" "Update" "Delete"]
+
+       "new subscription with empty existing modes in cache."
+       ["New" "Update" "Delete"]
+       []
+       ["New" "Update" "Delete"]
+
+       "Adding new subscription with existing modes."
```

Comment:
How about making the new-modes for this one just ["Delete"] so that not all the test cases are expected==new-modes

Reply:
I'm confused, this still seems to be expected outcome == new modes. I meant something like
expected ["Delete" "New"]
existing ["New"]
new ["Delete"]